azurerm/resource_arm_autoscale_setting.go

```diff
@@ -0,0 +1,954 @@
+														string(insights.Max),
+														string(insights.Min),
+														string(insights.Sum),
+													}, false),
+												},
+												"time_window": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: iso8601DurationString(),
+												},
+												"time_aggregation": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														string(insights.TimeAggregationTypeAverage),
+														string(insights.TimeAggregationTypeCount),
+														string(insights.TimeAggregationTypeMaximum),
+														string(insights.TimeAggregationTypeMinimum),
+														string(insights.TimeAggregationTypeTotal),
+													}, false),
+												},
+												"operator": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														string(insights.Equals),
+														string(insights.GreaterThan),
+														string(insights.GreaterThanOrEqual),
+														string(insights.LessThan),
+														string(insights.LessThanOrEqual),
+														string(insights.NotEquals),
+													}, false),
+												},
+												"threshold": {
+													Type:     schema.TypeInt,
+													Required: true,
+												},
+											},
+										},
+										Set: resourceAzureRmAutoscaleDefaultHash,
+									},
+									"scale_action": {
+										Type:     schema.TypeSet,
+										Required: true,
+										MinItems: 1,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"direction": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														string(insights.ScaleDirectionDecrease),
+														string(insights.ScaleDirectionIncrease),
+													}, false),
+												},
+												"type": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														string(insights.ChangeCount),
+														string(insights.PercentChangeCount),
+													}, false),
+												},
+												"value": {
+													Type:     schema.TypeString,
+													Required: true,
+												},
+												"cooldown": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: iso8601DurationString(),
+												},
+											},
+										},
+										Set: resourceAzureRmAutoscaleDefaultHash,
+									},
+								},
+							},
+						},
+						"fixed_date": {
+							Type:          schema.TypeSet,
+							Optional:      true,
+							MinItems:      1,
+							MaxItems:      1,
+							ConflictsWith: []string{"profile.recurrence"},
```

Comment:
Unfortunately not at this point - there's an issue open about this though. In general, some resources opt to handle this in the Create/Update methods - others just document the requirements.